### ISSUE TITLE
Store setter type signature allows setting readonly properties

### ISSUE THREAD

**Issue by JorrenH** [solidjs/solid#1359]
### Describe the bug

While creating a store using a getter property I encountered a runtime error trying to update the store. For example when I define the following store:

```ts
const [ store, setStore ] = createStore({
    get value() { return 'value' },
});

setStore('value', 'modified');
```

I would expect a type error on the last line, instead it results in a runtime error:
`Uncaught TypeError: setting getter-only property "value"`

### Your Example Website or App

https://stackblitz.com/edit/solidjs-templates-9qbvhq?file=src/App.tsx

### Steps to Reproduce the Bug or Issue

1. Go to the linked [stackblitz](https://stackblitz.com/edit/solidjs-templates-9qbvhq?file=src/App.tsx).
2. The app will compile
3. At runtime it will throw an error

### Expected behavior

As a user, the given code should produce a compile time type-error. Instead you get a runtime error.

### Platform

- OS: Windows (but affects others)
- Browser: Firefox (but affects others)
- Version: 106.0.5

**Comment by ryansolid** [comment]
Yeah makes sense. Honestly Store types are so complicated I have no idea how to even start approaching this one. But very much open to someone figuring it out.

**Comment by otonashixav** [comment]
I did start on this some time ago, though it breaks the type test cases that use generics. As such I don't intend to make a PR for this. Maybe it will be of help to someone else though.

https://github.com/otonashixav/solid/commit/72bff0da65fe19ec71cc923b29080b5fb14ffe97

**Comment by JorrenH** [comment]
Solved in #1444
